### PR TITLE
handle nested array config

### PR DESCRIPTION
**Pull request by grugna**
This is intended to fix a bug in guppy. With this change multiple nested array can be represented correctly.

### Bug fixes
- fixed a bug with resolving multi-level nested array fields

### CODE REVIEW

File: src/server/es/index.js [hunk 1]
```diff
@@ -226,11 +226,17 @@ class ES {
           }
           const fields = doc._source.array;
           fields.forEach((field) => {
-            if (!this.fieldTypes[index][field]) {
-              const errMsg = `[ES.initialize] wrong array entry from config: field "${field}" not found in index ${index}, skipped.`;
-              log.error(errMsg);
-              return;
-            }
+            const nodes = field.split('.');
+            var nestedObject = this.fieldTypes[index]
```
Comment (mfshao) [inline src/server/es/index.js:230]:
```suggestion
            let nestedObject = this.fieldTypes[index];
```

File: src/server/es/index.js [hunk 2]
```diff
@@ -226,11 +226,17 @@ class ES {
...
+            nodes.forEach((n) => {
+              if (n in nestedObject) {
+                nestedObject = nestedObject[n].properties;
+              } else {
+                const errMsg = `[ES.initialize] wrong array entry from config: field "${field}" not found in index ${index}, skipped.`;
+                console.log(errMsg);
+                return;
```
Comment (mfshao) [inline src/server/es/index.js:237]:
```suggestion
```

File: src/server/es/index.js [hunk 3]
```diff
@@ -226,11 +226,16 @@ class ES {
...
           const fields = doc._source.array;
           fields.forEach((field) => {
-            if (!this.fieldTypes[index][field]) {
-              const errMsg = `[ES.initialize] wrong array entry from config: field "${field}" not found in index ${index}, skipped.`;
-              log.error(errMsg);
-              return;
-            }
+            const nodes = field.split('.');
+            let nestedObject = this.fieldTypes[index];
+            nodes.forEach((n) => {
```
Comment (mfshao) [inline src/server/es/index.js:231]: I think you should just rewrite this loop through `nodes` using a plain for loop so you can break out if a `node` doesn't meet the creatia. You can't really break out or return easily from a `forEach` callback, so the `field` will stull be added

**Comment by ocshawn** [comment]
Please reopen and update if this is still needed, cleaning up old PRs.

**Pull request closed without merging**